motionplan/plannerOptions.go

```diff
@@ -186,6 +186,9 @@ type plannerOptions struct {
 	// profile is the string representing the motion profile
 	profile string
 
+	// planDeviationMM is a float representing how close we must be to the goal in order for planning to finish
+	planDeviationMM float64
```

Comment:
How is this distinct from GoalThreshold?

Reply:
let delta = abs(startPose - goalPose).
suppose we are on the first iteration of planning.
if delta <= planDevMM (should it exist) then we are done planning.

The difference is that GoalThreshold exists for all invocations of the planner, but planDevMM only exists for TPSpace RRT and is only used for the first iteration of planning.

Reply:
Why would we not want  the check `if delta <= GoalThreshold` for all planners?  In this case there is no work to be done you are already there

Reply:
I agree we don't want two separate values here. We should have the motion service pass the planDeviationMM as the goal threshold if we want to do the check.

Unsure at the moment if we want to pass the smaller or larger of the two numbers, thoughts?

Reply:
I think that is makes more sense to use the larger of the two numbers. 

At plan time, for a given moveBase call, we may already be within PlanDevMM of a goal but not within the goal threshold.

Reply:
>Unsure at the moment if we want to pass the smaller or larger of the two numbers, thoughts?

Oh, I see why you are unsure. The default value for GoalThreshold is 0.1. A user for instead could have their planDevMM set to 0.001. If we choose the max, then we are disobeying what the user wanted.

Reply:
However, on the flip side, lets say the user specified planDevMM == 5. Then if we choose the min, then we might end up constructing a plan and executing if even if we don't need to.

Reply:
oh silly me, instead of doing max/min of two values. Lets just set goalthreshold to be the value of planDevMM

Reply:
actually, I've convinced myself that planDevMM must be distinct from goalThreshold.

Consider the following scenario: We have a wheeled base which must go into a shed.
If we set goalThreshold to equal planDevMM then we may end up not inside. 
![IMG_2902](https://github.com/user-attachments/assets/e04474b2-8e11-4c67-8afe-d00e17057ca9)

Reply:
> I'm now wondering if perhaps we do after all want planDeviationMm to be fully handled in the motion service.

I am also thinking about this. As you mentioned on the merged PR related to this (https://github.com/viamrobotics/rdk/pull/4327#discussion_r1744238667) this should be handled within the motionplan package. If not, then there is no work to be done other than perhaps adding some comments around existing code. 

IMO, the only way to avoid the scenario where the base ends up outside the shed is to have plannerOptions include a field for planDevMM, and a supplementary field for planDevMM's success function only pertinent to the first iteration of our planner. 

This itself however, highlights an improper abstraction we've made which makes me wonder how else to approach the problem/issue.

Reply:
Okay, I'm looking more at the comment thread on that old PR, thinking about the behavior we've got, and here's what I've come up with.

The idea of `planDevMM` relies on the existence of a plan, and the fact that we may have deviated from it. It doesn't actually have anything to do with the generation of a plan.

Thinking along those lines, I think we should instead revisit the assumption that we should do nothing if we request a plan while being within `planDevMM` of the goal. Another interpretation would be simply that we would try to plan and move regardless, and this would simply disable replanning.

Reply:
Bumping this comment thread since I think it got missed and deciding the behavior here is core to what this PR should do @raybjork @nfranczak

Reply:
oop, I did miss this comment. Thank you for bumping the thread

Reply:
> The idea of planDevMM relies on the existence of a plan, and the fact that we may have deviated from it. It doesn't actually have anything to do with the generation of a plan.

This is a good point.

> I think we should instead revisit the assumption that we should do nothing if we request a plan while being within planDevMM of the goal. 

I think this is a safe assumption to be made. Suppose we are within planDevMM of the goal at start and we've generated some plan whose inputs are non-zero. Then executing that generated plan is pointless. 



> Another interpretation would be simply that we would try to plan and move regardless, and this would simply disable replanning.

I do not think this is safe. If the kinematic base driver is faulty for some reason we will end up executing the plan and might wind up somewhere that is outside of the defined planDevMM. This would then cause for a replan to be issued and we might end up stuck in a cycle of perpetual replanning. 


Overall, I think the best way to handle this is to return a plan whose inputs are: `[0, 0, 0, 0], [0, 0, 0, 0]`.


@raybjork what are your thoughts on this?

Reply:
Sorry I missed this earlier.  

>The idea of planDevMM relies on the existence of a plan, and the fact that we may have deviated from it. It doesn't actually have anything to do with the generation of a plan.

Yes I agree with this it has to do with tracking the plan but not the generation of it. 

Therefore it feels to me like the behavior should be the same - always requesting a plan, and then once it is returned the execution begins.  In execution we see that we are within the goal region and should return a "success".  The optimization of not requesting a plan if we never intend on following it seems really minor at the moment.

Reply:
> The optimization of not requesting a plan if we never intend on following it seems really minor at the moment.

ok, in that case I think this ticket should be closed and marked as "won't do".
@raybjork @biotinker would you agree?